packages/react-dom/src/InView.tsx

```diff
@@ -0,0 +1,141 @@
+import { Component, createElement } from 'react'
+import { observe } from './observe'
+import type { IntersectionObserverProps, PlainChildrenProps } from './types'
+
+function isPlainChildren(props: IntersectionObserverProps | PlainChildrenProps): props is PlainChildrenProps {
+  return typeof props.children !== 'function'
+}
```

Comment:
The usage of isPlainChildren is singular and it's used as a negation, as shown below:

```tsx
 if (!isPlainChildren(this.props)) {
      // Store the current State, so we can pass it to the children in the next render update
      // There's no reason to update the state for plain children, since it's not used in the rendering.
      this.setState({ inView, entry })
    }
```

It's a minor suggestion, but how about using isFunctionChildren instead? I don't think there's a need to use a negation for isPlainChildren in this case.

Reply:
Oh! cool! I removed isPlainChildren. 

```diff
  handleChange = (inView: boolean, entry: IntersectionObserverEntry) => {
    if (inView && this.props.triggerOnce) {
      // If `triggerOnce` is true, we should stop observing the element.
      this.unobserve()
    }
-    if (!isPlainChildren(this.props)) {
+   if (this.props.children === 'function') {
      // Store the current State, so we can pass it to the children in the next render update
      // There's no reason to update the state for plain children, since it's not used in the rendering.
      this.setState({ inView, entry })
    }
    if (this.props.onChange) {
      // If the user is actively listening for onChange, always trigger it
      this.props.onChange(inView, entry)
    }
  }
```